### DISCUSSION TITLE
How to store encoded_nodata ?

### DISCUSSION THREAD

**Discussion by remi-braun** [corteva/rioxarray#313] [Q&A]
I did an addition between two xarrays and as seen in [the example](https://corteva.github.io/rioxarray/stable/getting_started/manage_information_loss.html) I lost the attributes (except the CRS, I don't know if its normal)
I managed to set them back as stated, except the `encoded_nodata`:
```python
# xda: xr.Dataarray
xda_out = xda + xda
xda_out.rio.write_crs(xda.rio.crs, inplace=True)   # OK
xda_out.rio.update_attrs(xda.rio.attrs, inplace=True)  # OK

# this line is not shown in the example, sets nodata = nan but forgets encoded_nodata
xda_out.rio.write_nodata(xda.rio.nodata, inplace=True) 
```

How do I achieve that ? Because I work with nodata pixels == `nan` and I need to write the correct value back when I save the raster to disk.

**Comment by remi-braun** [comment]
It works if I state this: 

```python
xda_out.encoding = xda.encoding
```
But isn't it a bit complicated to write 4 lines of code to get back the raster metadata integrity after a simple operation ?
Does a function doing all this exist ?

**Reply by snowman2** [reply]
http://xarray.pydata.org/en/stable/generated/xarray.set_options.html#xarray-set-options

You could try `keep_attrs=True`

**Reply by remi-braun** [reply]
Oh this could have been nice.
Unfortunately it doesn't keep nodata values 😢 

I'll wrap these lines in a function then :)

**Comment by remi-braun** [comment]
Another question about this `encoded_nodata` value.
If I want to properly set the nodata on an array with the `mask` style, I must do:

```python
    # xda: some xr.DataArray without set any nodata
    xda_nodata = xda.where(xda !=nodata_val)
    xda_nodata.rio.write_nodata(None, inplace=True)
    xda_nodata.encoding["_FillValue"] = nodata_val
```

Is there another shortcut ?

**Reply by snowman2** [reply]
> Is there another shortcut ?

Not currently. Might not be bad to have a flag in `write_nodata` for when the nodata value is encoded.
```python
xda_nodata.rio.write_nodata(nodata_val, encoded=True)
```

**Reply by remi-braun** [reply]
I was wondering if the function `write_nodata` should or not set the nodata pixels to `np.nan` in case `encoded=True`.
Right now I think this could be great, but there might be some downfall I don't see.

**Reply by snowman2** [reply]
See test in #314. It returns `nan` if `rio.encoded_nodata` is not None. So, no need to write it.

**Reply by remi-braun** [reply]
When do you plan to deliver your next release ? 😄

**Reply by snowman2** [reply]
Current plan is to wait until after xarray 0.18 is released due to the entrypoint addition (https://github.com/pydata/xarray/issues/4697).
